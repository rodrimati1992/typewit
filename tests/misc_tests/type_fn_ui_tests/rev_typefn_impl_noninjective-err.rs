use typewit::{RevTypeFn, TypeFn};

struct NonInjective;

impl<T> TypeFn<T> for NonInjective {
    type Output = u8;

    const TYPE_FN_ASSERTS: () = { let _: typewit::CallInjFn<Self, T>; };
}

impl RevTypeFn<u8> for NonInjective {
    type Arg = ();
}


fn main(){}