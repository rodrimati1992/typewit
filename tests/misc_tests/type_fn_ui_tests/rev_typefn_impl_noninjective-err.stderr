error[E0271]: type mismatch resolving `<NonInjective as RevTypeFn<u8>>::Arg == T`
  --> tests/misc_tests/type_eq_ui_tests/rev_typefn_impl_noninjective-err.rs:8:67
   |
5  | impl<T> TypeFn<T> for NonInjective {
   |      - this type parameter
...
8  |     const TYPE_FN_ASSERTS: () = { let _: typewit::CallInjFn<Self, T>; };
   |                                                                   ^ type mismatch resolving `<NonInjective as RevTypeFn<u8>>::Arg == T`
   |
note: expected this to be `T`
  --> tests/misc_tests/type_eq_ui_tests/rev_typefn_impl_noninjective-err.rs:12:16
   |
12 |     type Arg = ();
   |                ^^
   = note: expected type parameter `T`
                   found unit type `()`
   = note: required for `NonInjective` to implement `InjTypeFn<T>`
